internal/rootcoord/meta_table.go

```diff
@@ -22,32 +22,28 @@ import (
 	"fmt"
 	"sync"
 
-	"github.com/milvus-io/milvus/internal/metrics"
-
-	"github.com/milvus-io/milvus/internal/util/timerecord"
-
-	"github.com/milvus-io/milvus/internal/common"
-
-	pb "github.com/milvus-io/milvus/internal/proto/etcdpb"
-
 	"go.uber.org/zap"
 
 	"github.com/milvus-io/milvus-proto/go-api/milvuspb"
+	"github.com/milvus-io/milvus/internal/common"
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/metastore"
 	"github.com/milvus-io/milvus/internal/metastore/model"
+	"github.com/milvus-io/milvus/internal/metrics"
+	pb "github.com/milvus-io/milvus/internal/proto/etcdpb"
 	"github.com/milvus-io/milvus/internal/proto/internalpb"
 	"github.com/milvus-io/milvus/internal/util/contextutil"
 	"github.com/milvus-io/milvus/internal/util/funcutil"
+	"github.com/milvus-io/milvus/internal/util/timerecord"
 	"github.com/milvus-io/milvus/internal/util/typeutil"
 )
 
-//go:generate mockery --name=IMetaTable --outpkg=mockrootcoord
+//go:generate mockery --name=IMetaTable --outpkg=mockrootcoord --filename=meta_table.go --with-expecter
 type IMetaTable interface {
 	AddCollection(ctx context.Context, coll *model.Collection) error
 	ChangeCollectionState(ctx context.Context, collectionID UniqueID, state pb.CollectionState, ts Timestamp) error
 	RemoveCollection(ctx context.Context, collectionID UniqueID, ts Timestamp) error
-	GetCollectionByName(ctx context.Context, collectionName string, ts Timestamp) (*model.Collection, error)
+	GetCollectionByName(ctx context.Context, collectionName string, ts Timestamp, allowUnavailable bool) (*model.Collection, error)
```

Comment:
I think we should always filter unavailable for these requests by collection name.

Reply:
It just gets a collection meta by collection name,  and it is similar to the GetCollectionByID method